packages/edit-post/src/components/resize-canvas/index.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * WordPress dependencies
+ */
+import { useSimulatedMediaQuery } from '@wordpress/block-editor';
+import { useSelect } from '@wordpress/data';
+import { useEffect, useState } from '@wordpress/element';
+
+/**
+ * Function to resize the editor window.
+ *
+ * @return {Object} Inline styles to be added to resizable container.
+ */
+export function useResizeCanvas() {
+	const deviceType = useSelect( ( select ) => {
+		return select( 'core/block-editor' ).getPreviewDeviceType();
+	}, [] );
+
+	const [ actualWidth, updateActualWidth ] = useState( window.innerWidth );
+
+	useEffect( () => {
+		const resizeListener = () => updateActualWidth( window.innerWidth );
+		window.addEventListener( 'resize', resizeListener );
+
+		return () => {
+			window.removeEventListener( 'resize', resizeListener );
+		};
+	} );
+
+	const getCanvasWidth = ( device ) => {
+		let deviceWidth = 0;
+
+		switch ( device ) {
+			case 'Tablet':
+				deviceWidth = 780;
```

Comment:
Could have a filter or setting to be able to modify these values? They will change per theme and per client as different designs require different breakpoints.

Reply:
@mikeselander I just noticed your comments. I definitely recommend creating an issue.

I just happened to be looking at this PR while triaging https://github.com/WordPress/gutenberg/issues/21235, which seems relevant to your comments.